### PR TITLE
Fix implicit conversion warnings

### PR DESCRIPTION
**Pull request by drizt**

### CODE REVIEW

File: src/zip.c [hunk 1]
```diff
@@ -699,7 +699,7 @@ int zip_entry_fread(struct zip_t *zip, const char *filename) {
 
   xattr = (info.m_external_attr >> 16) & 0xFFFF;
   if (xattr > 0) {
-    if (chmod(filename, xattr) < 0) {
+    if (chmod(filename, (int)xattr) < 0) {
```
Comment (kuba--) [inline src/zip.c]: cast to `mode_t`

File: src/zip.c [hunk 2]
```diff
@@ -871,7 +871,7 @@ int zip_extract(const char *zipname, const char *dir,
 #else
     xattr = (info.m_external_attr >> 16) & 0xFFFF;
     if (xattr > 0) {
-      if (chmod(path, xattr) < 0) {
+      if (chmod(path, (int)xattr) < 0) {
```
Comment (kuba--) [inline src/zip.c]: cast to `mode_t`

File: src/zip.c [hunk 3]
```diff
@@ -160,7 +160,7 @@ struct zip_t *zip_open(const char *zipname, int level, char mode) {
   case 'a':
     if (!mz_zip_reader_init_file(
             &(zip->archive), zipname,
-            level | MZ_ZIP_FLAG_DO_NOT_SORT_CENTRAL_DIRECTORY)) {
+            (mz_uint)level | MZ_ZIP_FLAG_DO_NOT_SORT_CENTRAL_DIRECTORY)) {
```
Comment (kuba--) [inline src/zip.c]: I think here would be better to use: `zip->level`

File: src/zip.c [hunk 4]
```diff
@@ -51,7 +51,7 @@
     }                                                                          \
   } while (0)
 
-static char *basename(const char *name) {
+static const char *basename(const char *name) {
```
Comment (kuba--) [inline src/zip.c]: 👍

File: src/zip.c (hunk 1 above)
Comment (drizt) [reply src/zip.c]: On MinGW it takes `int`

File: src/zip.c (hunk 3 above)
Comment (drizt) [reply src/zip.c]: Agree

File: src/zip.c (hunk 1 above)
Comment (kuba--) [reply src/zip.c]: `mode_t` is not defined on MinGW?
I hop good thing about `mode_t` is it differs per platform/compiler.

Comment (kuba--) [reply src/zip.c]: if `mode_t` is  not defined on compiler/platform we need to define `mode_t` as an alias first.

Comment (drizt) [reply src/zip.c]: Replaced with `mode_t`. MinGW actually has `mode_t`. And `mode_t` doesn't produce warnings.

**Pull request merged**